### ISSUE TITLE
Collecting Issue

### ISSUE THREAD

**Issue by TemhAAhmeT** [Th3C0D3R/FoBJS_Release#11]
Hi,
I noticed that bot doesn't collect all gold buildings. I have approx. 30 gold producing building and bot randomly ignores 8-10 buildings.
When bot collects all, close the bot and log in via browser and there are some gold remains there.
Collect them manually and wait for all buildings production (5min) and run the bot and again close the bot.
Also i noticed that by calculating my production for few hours. It is approx. %30-%40 less then i expected.

**Comment by Th3C0D3R** [comment]
Can you name those buildings please are those always the same or is it random?

**Comment by TemhAAhmeT** [comment]
Its Hut(Stone Age)
Its definitely same buildings (12 of 34 buildings always remain uncollected)

**Comment by Th3C0D3R** [comment]
I will look into it as soon as possible.

Is every hut not collected or just some?

**Comment by TemhAAhmeT** [comment]
12 of 34 buildings. Always the same 12.

**Comment by Th3C0D3R** [comment]
Uhm did the wrong question:
Is every not collected building a hut?

**Comment by TemhAAhmeT** [comment]
Yes all are hut but i do have only huts for gold. So not sure about other buildings.

**Comment by Th3C0D3R** [comment]
Ok. i will look into It as soon as possible

**Comment by TemhAAhmeT** [comment]
Ok, thanks for your time.

**Comment by Th3C0D3R** [comment]
Soo after some investigation I thinkt the problem is:
Because I group all Productiontypes (Goods, Supplies, Residential Buildings) togather, if they have a nearly same finishtime, they bot trys to collect buildings which will be finished a short time after that (1-x seconds) sooo I look what I can do

**Comment by TemhAAhmeT** [comment]
Good luck :D

**Comment by TemhAAhmeT** [comment]
Hi,
I tried 3 times.
2nd try all buildings (gold and production) worked/clicked perfectly.
1st and 3rd tries, 7-10 gold buildings are passed/not clicked.

**Comment by TemhAAhmeT** [comment]
Any update?

**Comment by Th3C0D3R** [comment]
I am currently working on an other project regarding foe-helper

I started to work on a new and rewritten version of the bot. But it will need some time after the mentioned first project

**Comment by TemhAAhmeT** [comment]
Oke, good luck.
So probably few months.
Will stop checking your repo regularly :D than ...

**Comment by TemhAAhmeT** [comment]
After the update bot cant even login. Deleting AppData etc. doesnt help.
I know you are not working on this right now but a small update would be awesome.
Cheers.

**Comment by Th3C0D3R** [comment]
There it is 😄

**Comment by TemhAAhmeT** [comment]
![Untitled](https://user-images.githubusercontent.com/12784047/80638495-cf70ef80-8a60-11ea-9443-5c47496fe55d.png)
![Untitled](https://user-images.githubusercontent.com/12784047/80638571-ee6f8180-8a60-11ea-8bfa-08d563de6607.jpg)


There is some bug keeps bugging.
Also of course i have deleted everything before new version.

**Comment by Th3C0D3R** [comment]
Is the production bot active? If so, please deactivate it

**Comment by TemhAAhmeT** [comment]
Yes, production bot active. And the error comes at the start of collecting.
Production bot is not perfect but i found how to use it. So i was using it at version 1.2.3 and there was no error message.

**Comment by Th3C0D3R** [comment]
please dont use the production bot. its not working perfectly and I dont know what will happen if there are to many errors (on InnoGames side e.g Bans)

**Comment by TemhAAhmeT** [comment]
I am producing tools at 5 min intervals and gold for 1 hour buildings. And its collecting all gold eventually. I mean i have a way to collect all with production bot. So i was using production bot all the time. Months i mean. So i dont think there is a risk.
Just this error message is annoying. But i will keep using the production bot. It would be so nice if you make disappear that error message. If you can spare little more of your time.

**Comment by Th3C0D3R** [comment]
the error message comes because the way the bot works in the tool is 100% inefficent and total bad. maybe I will work on the new bot version and do the productionbot way better. Currently I am unable to fix the issue.

**Comment by TemhAAhmeT** [comment]
Ok, no problem. If its not fixable, i can get used the that messages :D.
Nevertheless thanks for your time.
stay home stay healthy :D

**Comment by Th3C0D3R** [comment]
Thanks,
Same for you ✌🏻
